lib/spotify_uri_bot/api.ex

```diff
@@ -34,88 +37,53 @@ defmodule SpotifyUriBot.Api do
     else
       err ->
         Logger.error("Get token failed with error: #{inspect(err)}")
+        Logger.error("Retrying...")
+        Process.sleep(500)
```

Comment:
This will block the caller, so if it start failing you could block the caller by an infinite time, and that's not great... There are times that you really need to return an error.

Maybe with a counter and just try N times

(This applies to all the Process.sleep with retries)

Reply:
That is very true, yesterday I got that same problem while testing, so I will definitely change it. Thanks a lot for the review! It means a lot :purple_heart:

Reply:
You will always have my reviews ;-)

You could do something like a `retry_n` function like:

``` elixir
defp retry_n(n, callback) when n <= 0, do: callback.()
defp retry_n(n, callback) do
  case callback.() do
    {:error, _} -> retry_n(n - 1, callback)
    result -> result
  end
end 
```